app/scripts/migrations/fail-tx.js

```diff
@@ -0,0 +1,41 @@
+const clone = require('clone')
+
+module.exports = function (version, reason, condition) {
+  return function (originalVersionedData) {
+    const versionedData = clone(originalVersionedData)
+    versionedData.meta.version = version
+    try {
+      const state = versionedData.data
+      const newState = transformState(state, condition, reason)
+      versionedData.data = newState
+    } catch (err) {
+      console.warn(`MetaMask Migration #${version}` + err.stack)
+    }
+    return Promise.resolve(versionedData)
+
+  }
+}
+
+function transformState (state, condition, reason) {
+  const newState = state
+  const { TransactionController } = newState
+  if (TransactionController && TransactionController.transactions) {
+    const transactions = newState.TransactionController.transactions
```

Comment:
`newState.TransactionController` can be just `TransactionController` since we've destructured `newState` above (similarly for the line below)

Reply:
Addressed in new commit.